packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2885,7 +2876,27 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     );
     return prunedGraphData;
   };
-
+  private createBulkServiceRegistrationInput = (
+    graph: PureModel,
+    services: Service[],
+  ): ServiceRegistrationInput[] => {
+    const graphData = this.getFullGraphModelData(graph);
+    const results: ServiceRegistrationInput[] = [];
+    services.forEach((service) => {
+      const prunedGraphData = this.prunePureModelContextData(
+        graphData,
+        (element: V1_PackageableElement) => !(element instanceof V1_Service),
+      );
+      prunedGraphData.elements = graphData.elements.filter(
```

Comment:
you shouldnt need this line anymore. 
covered by the function in your prune method 
```
 const prunedGraphData = this.prunePureModelContextData(
        graphData,
        (element: V1_PackageableElement) => !(element instanceof V1_Service),
      );
```